### PR TITLE
Add userdata secret to vm template

### PR DESCRIPTION
**Pull request by chrisho**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Related Issue:**
https://github.com/harvester/harvester/issues/2932

### CODE REVIEW

File: pkg/data/template.go [hunk 1]
```diff
@@ -80,6 +107,32 @@ func initBaseTemplateVersion(vmTemplateVersions ctlharvesterv1.VirtualMachineTem
...
+  labels:
+    harvesterhci.io/cloud-init-template: harvester
+type: secret
+data:
+  networkdata: ""
+  userdata: I2Nsb3VkLWNvbmZpZwpwYWNrYWdlX3VwZGF0ZTogdHJ1ZQpwYWNrYWdlczoKICAtIHFlbXUtZ3Vlc3QtYWdlbnQKcnVuY21kOgogIC0gLSBzeXN0ZW1jdGwKICAgIC0gZW5hYmxlCiAgICAtIC0tbm93CiAgICAtIHFlbXUtZ3Vlc3QtYWdlbnQuc2VydmljZQo=
+---
+apiVersion: v1
+kind: Secret
+metadata:
```
Comment (futuretea) [inline pkg/data/template.go:150]: maybe need ownerefence？

**Review by futuretea — APPROVED** [review]
LGTM

**Review by guangbochen — APPROVED** [review]
LGTM, verified that both the iso and raw template have the `quem-guest-agent` config checked by default, also with RBAC, both the project owner and member have permission to access the cloud-init data.

**Review by Vicente-Cheng — APPROVED** [review]
tested with upgrading and fresh install.
Upgrade looks well.
For a fresh installation, both templates have the correct configs.

LGTM!

**Review by bk201 — APPROVED** [review]

**Pull request merged**